unpack-trees.c

```diff
@@ -1726,7 +1726,8 @@ int unpack_trees(unsigned len, struct tree_desc *t, struct unpack_trees_options
 			if (git_env_bool("GIT_TEST_CHECK_CACHE_TREE", 0))
 				cache_tree_verify(the_repository, &o->result);
 			if (!cache_tree_fully_valid(o->result.cache_tree))
-				cache_tree_update(&o->result,
+				cache_tree_update(the_repository,
```

Comment:
Here, too, we're outside of `builtin/` and outside of `t/helper/`, so it is not really okay to _introduce_ `the_repository`.

In this instance, we should consider putting it into `struct unpack_trees_options`, anyway, but that will require a much larger set of changes.

Reply:
I've made the changes in other files. Changing this would need changes to a lot of files. What would you suggest? Should I make the changes to those files?

Reply:
It is ultimately up to you: this is a rabbit hole, and it is your decision how far down you want to go.

In general, I would suggest to do the conversion one function signature at a time. All the callers that do _not_ have an `r` to work with should get it via a (transitional!) `struct repository *r = the_repository;` at the beginning of the function, just like you did here: https://github.com/gitgitgadget/git/pull/915/commits/c977a2aaa0a5e5046c3f7d596d8076743b456776#diff-eeb5ad245cf301bdd1dcd3253bf45156af5171a7dd82abf60ed241a044d704d4R65

Ultimately, we will want to have the entire code base converted, with `cmd_*()` functions in `builtin/*.c` getting their `struct repository *r` passed in as parameter.

But that is probably _many, many_ years down the road. It's also a pretty disruptive set of changes, so it also makes sense to pay attention to what is happening in Git's `seen` branch, which is a branch collecting most of the patch series that are currently in flight. If your patches conflict with any of those, it is a safe bet that it would be wise to work on other parts of the code ;-)

Reply:
> In general, I would suggest to do the conversion one function signature at a time. All the callers that do _not_ have an `r` to work with should get it via a (transitional!) `struct repository *r = the_repository;` at the beginning of the function, just like you did here: [c977a2a#diff-eeb5ad245cf301bdd1dcd3253bf45156af5171a7dd82abf60ed241a044d704d4R65](https://github.com/gitgitgadget/git/commit/c977a2aaa0a5e5046c3f7d596d8076743b456776#diff-eeb5ad245cf301bdd1dcd3253bf45156af5171a7dd82abf60ed241a044d704d4R65)
> 
Okay, I'll try to do this and convert them one at a time for this patch and thereby address individual callers in different patches.

> But that is probably _many, many_ years down the road. It's also a pretty disruptive set of changes, so it also makes sense to pay attention to what is happening in Git's `seen` branch, which is a branch collecting most of the patch series that are currently in flight. If your patches conflict with any of those, it is a safe bet that it would be wise to work on other parts of the code ;-)

Yes, I'll be on the lookout for those :). Thanks for helping. :smile:

Reply:
Yes. This is totally legitimate, as we want to convert the code base _incrementally_, to make it substantially easier to review.